distro.go

```diff
@@ -191,24 +218,23 @@ configuration: %v
 	}
 
 	// Generate the string
-	return fmt.Sprintf(`distro: %s
-configuration:
+	return fmt.Sprintf(`configuration:
   - Version: %d
   - DefaultUID: %d
   - InteropEnabled: %t
   - PathAppended: %t
   - DriveMountingEnabled: %t
   - undocumentedWSLVersion: %d
-  - DefaultEnvironmentVariables:%s`, d.Name(), c.Version, c.DefaultUID, c.InteropEnabled, c.PathAppended,
+  - DefaultEnvironmentVariables:%s`, c.Version, c.DefaultUID, c.InteropEnabled, c.PathAppended,
 		c.DriveMountingEnabled, c.undocumentedWSLVersion, fmtEnvs)
 }
```

Comment:
Actually, now that I saw that, I see that it was in String() before :) Which made more sense to me for the above reason (apart if you reused some of those private methods). The methos is now really long in the end.

Reply:
Yeah, it used to be all in String but I refactored them out for two reasons:
- GUID, Config and the Name formatting all in the same function was a bit tedious to read
- Having them in separate functions allows for returning early on error.

Honestly I belive that as it is, it's too verbose. I didn't want to mix apples and oranges in this PR, but this:
```
distro{Name: Ubuntu, GUID: {01234567-0123-0123-0123-0123456789ab} }
```
seems more reasonable than:
```
name: Ubuntu
guid: {01234567-0123-0123-0123-0123456789ab}
configuration:
  - Version: 2
  - DefaultUID: 0
  - InteropEnabled: true
  - PathAppended: true
  - DriveMountingEnabled: true
  - undocumentedWSLVersion: 2
  - DefaultEnvironmentVariables:
    - HOSTTYPE: x86_64
    - LANG: en_US.UTF-8
    - PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
    - TERM: xterm-256color
```

I'd even venture even further to say the `String` method should only print the name, since the GUID and Config may change over time or even not exist at all. We can implement `(Configuration).String() string` or `(Distro).Info() string` separately if we see the need at some point.

Reply:
If we decide to keep all this info, I'll refactor it to use the `yaml`package.

Reply:
Agreed that `String` should not report the config, name/gui is enough to define it. (I would still prints the GUID as it can be useful to differentiate removed-but-still-in-memory ones). But that’s not part of this PR, as you said.

> Having them in separate functions allows for returning early on error.
This is the deal breaker to me and then, agreed for the split.